src/CsProj/ProjectFileParser.cs

```diff
@@ -41,6 +43,27 @@ select prop
                 select prop
             ).FirstOrDefault();
             PackageVersion = xPackageVersion?.Value ?? Version;
+
+            var packageId = (
+                from prop in propertyGroup.Elements()
+                where prop.Name == "PackageId"
+                select prop
+            ).FirstOrDefault();
+
+            var title = (
+                from prop in propertyGroup.Elements()
+                where prop.Name == "Title"
+                select prop
+            ).FirstOrDefault();
+            PackageName = title?.Value ?? packageId.Value;
+
+            if (string.IsNullOrEmpty(PackageName))
```

Comment:
It would preferible to load each field separately?

Something like 
```c#
_fileParser.Load(csProjXml, ProjectFileProperty.Version);
```

`ProjectFileProperty.Version` could be an enum in `CsProj` folder.

```c#
public class ProjectFileParser
{
   ...
   public virtual void Load(string xmlDocument, ProjectFileProperty property)
   {
       var xml = XDocument.Parse(xmlDocument, LoadOptions.PreserveWhitespace);

      // Project should be root of the document
      var project = xml.Elements("Project");
      var xProject = project as IList<XElement> ?? project.ToList();
      if(!xProject.Any())
      {
         throw new ArgumentException(
            "The provided csproj file seems malformed - no <Project> in the root", 
            paramName: nameof(xmlDocument)
         );
      }

      var propertyGroup = xProject.Elements("PropertyGroup");

      switch(property) {
         case ProjectFileProperty.Version: 
            // TODO
            break;
         case ProjectFileProperty.PackageVersion: 
            // TODO
            break;
         ...
      }
      ...
   }
...
}
```

Reply:
Yes I guess that would work, as we then only invoke your new properties if a message is carrying a custom format?

Reply:
Exactly, that is what I was thinking.

Later today I will upload a new commit with this change so we can review it

Reply:
I've uploaded this change, so you can take a look at it whenever you can